micro-rdk/src/common/conn/viam.rs

```diff
@@ -390,6 +391,85 @@ where
         }
     }
 }
+
+pin_project_lite::pin_project! {
```

Comment:
don't you need to use the `#[pin]` attribute on the specific fields you want pinned?